go/minterm/minterm.go

```diff
@@ -37,6 +40,7 @@ func New() (*MinTerm, error) {
 
 // Shutdown closes the terminal.
 func (m *MinTerm) Shutdown() error {
+	m.restore()
```

Comment:
Can `Shutdown` be called from a different goroutine than `readLine`?  If so, we can race over `m.raw`